### PR TITLE
Refactor update action in nodes controller (bnc#918407)

### PR DESCRIPTION
**Pull request by vwallfahrer**
https://bugzilla.suse.com/show_bug.cgi?id=918407

### CODE REVIEW

File: crowbar_framework/app/controllers/nodes_controller.rb [hunk 1]
```diff
@@ -338,25 +338,49 @@ def edit
   end
 
   def update
-    unless request.post?
-      raise ActionController::UnknownHttpMethod.new("POST is required to update proposal #{params[:id]}")
-    end
+    respond_to do |format|
+      format.json do
+        unless request.post?
```
Comment (tboerger) [inline crowbar_framework/app/controllers/nodes_controller.rb:343]: That part should be dropped and needs to be done through the routes.

**Comment by rhafer** [comment]
Does this really address some bug in bugzilla? If yes, could you please fix the bnc# reference in the title?

File: crowbar_framework/app/controllers/nodes_controller.rb (hunk 1 above)
Comment (MaximilianMeister) [inline crowbar_framework/app/controllers/nodes_controller.rb:343]: wouldn't it be better to have some nice answer here, instead of an exception on the server side, and a site not found in html?

Comment (tboerger) [inline crowbar_framework/app/controllers/nodes_controller.rb:343]: It would be nicer if this API endpoint just accepts post instead of everything :)

Comment (tboerger) [inline crowbar_framework/app/controllers/nodes_controller.rb:343]: But as this have been like that before let's just keep that for another PR

**Comment by rhafer** [comment]
Given the bug report is against Cloud 6 I am pretty sure that this one was supposed to be done against master first (and then probably backported).

**Comment by tboerger** [comment]
This have also been done for master.
